### PR TITLE
PBI-Chore: Setup logger

### PR DESCRIPTION
**Pull request by valenciusap17**
## Describe your changes
- Setup logger
- Integrate current code with new logger

## Patterns for adding the logger (open to discussion)
- Info: Point a significant event on while application is running, e.x starting of an application, request received on router
- Debug: Printing some outputs for troubleshooting purposes  
- Error: Used when some function gone wrong but not harming the application running process
- Fatal: Placed when some failures appear that may harm the data or the running process of the application

### CODE REVIEW

**Comment by valeelim** [comment]
I think the tests are failing @valenciusap17

File: cmd/main.go [hunk 1]
```diff
@@ -15,12 +15,25 @@ import (
...
+	logger.Debug("Printing stuff")
+	logger.Error("Error occured")
+	// logger.Panic("Panic occured")
+	// logger.Fatal("Fatal occured")
+	// u.GeneralLogger.Println("Starting...")
+
+	u.GeneralLogger.Println("Loading configurations")
 	cfg := config.Load()
 
+	u.GeneralLogger.Println("Creating database connection")
```
Comment (valeelim) [inline cmd/main.go:36]: I don't think we need these loggers since they will just panic if they fail anyways

File: log/general-log.log [hunk 2]
```diff
@@ -0,0 +1,54 @@
+General Logger:	2024/11/01 13:56:31 main.go:24: Starting...
```
Comment (valeelim) [inline log/general-log.log:1]: don't push this file 😂

File: cmd/main.go [hunk 3]
```diff
@@ -15,12 +15,25 @@ import (
...
 // main is the entrypoint for the entire service
 func main() {
+	logger := u.NewConsoleLogger()
+
+	logger.Info("Starting...")
+	logger.Debug("Printing stuff")
+	logger.Error("Error occured")
+	// logger.Panic("Panic occured")
+	// logger.Fatal("Fatal occured")
+	// u.GeneralLogger.Println("Starting...")
```
Comment (edutjie) [inline cmd/main.go:31]: Don't forget to remove unused comments

File: cmd/utils/base_logger.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+var GeneralLogger *log.Logger
+var ErrorLogger *log.Logger
+var DebugLogger *log.Logger
+var InfoLogger *log.Logger
+var PanicLogger *log.Logger
+var FatalLogger *log.Logger
+
+// TODO
+// Use builder pattern to create a more descriptive logging functions
```
Comment (edutjie) [inline cmd/utils/base_logger.go:18]: Minor, but use `// TODO: <message>` so it will have different color on our IDE

**Comment by ferryhieronimus** [comment]
Could you elaborate more on this PR? How do we use this logger?

File: cmd/utils/base_logger.go [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	"os"
+	"path/filepath"
+)
+
+var errorLogger *log.Logger
+var debugLogger *log.Logger
+var infoLogger *log.Logger
+var fatalLogger *log.Logger
+
+func init() {
```
Comment (valeelim) [inline cmd/utils/base_logger.go:15]: what is this file for?

**Comment by valeelim** [comment]
LGTM, there's a conflict though

**Review by valeelim — APPROVED** [review]

**Pull request merged**